File: sdk/src/main/java/com/meniga/sdk/webservices/requests/AcceptChallenge.java
package com.meniga.sdk.webservices.requests;

import com.meniga.sdk.helpers.MenigaDecimal;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */
public class AcceptChallenge extends QueryRequestObject {

	public UUID id;
	public MenigaDecimal targetAmount;

	@Override
	public long getValueHash() {
		int result = id != null ? id.hashCode() : 0;
		result = 31 * result + (targetAmount != null ? targetAmount.hashCode() : 0);
		return result;
	}


	@Override
	public Map<String, String> toQueryMap() {
		return new HashMap<>();
	}
}

File: sdk/src/main/java/com/meniga/sdk/webservices/challenge/AcceptChallenge.kt
/*
 * Copyright 2017-2018 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.challenge

import com.meniga.sdk.helpers.MenigaDecimal
import com.meniga.sdk.webservices.requests.QueryRequestObject
import java.util.UUID

/**
 * @param id The id of the challenge to accept.
 * @param targetAmount The target amount for the challenge. If not specified, the target amount will be calculated according to the target percentage of the challenge.
 */
internal data class AcceptChallenge(
        @JvmField var id: UUID? = null,
        @JvmField var targetAmount: MenigaDecimal? = null
) : QueryRequestObject() {

    override fun getValueHash(): Long = hashCode().toLong()
}
